skyplane/api/transfer_job.py

```diff
@@ -158,8 +158,15 @@ def _run_multipart_chunk_thread(
                     region = dest_iface.region_tag()
                     dest_object = dest_objects[region]
                     _, upload_id = upload_id_mapping[region]
+
+                    custom_data = None
+                    # Convert parts to base64 and store mime_type if destination interface is AzureBlobInterface
+                    if isinstance(dest_iface, AzureBlobInterface):
+                        block_ids = list(map(lambda part_num: AzureBlobInterface.id_to_base64_encoding(part_num), parts))
```

Comment:
You can think of `id_to_base64_encoding` as providing an "agreed-upon" mapping of part numbers to block IDs (the format Azure accepts - same length + base64 encoded) between the gateway operators and the local transfer_job.py code. Instead of special casing Azure and sending the blockIDs as the `part_number` of a Chunk to the gateways, and modifying the portion of the code that is shared by every cloud provider, I decided to use this static method to represent a deterministic (part_number) -> (Azure blockID) mapping and provide this list of block_ids as a `custom_data` to the transfer list. (maybe in the future, we use this custom data field for different cloud providers as well - Lambda, BackBlaze, IBMCloud etc.)